src/renderers/native/ReactNativeFiber.js

```diff
@@ -0,0 +1,378 @@
+  },
+
+  commitUpdate(
+    instance : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    rootContainerInstance : Object,
+    internalInstanceHandle : Object
+  ) : void {
+    const viewConfig = ReactNativeViewConfigRegistry.get(type);
+
+    precacheFiberNode(internalInstanceHandle, instance._nativeTag);
+
+    const updatePayload = ReactNativeAttributePayload.diff(
+      oldProps,
+      newProps,
+      viewConfig.validAttributes
+    );
+
+    UIManager.updateView(
+      (instance : any)._nativeTag, // reactTag
+      viewConfig.uiViewClassName, // viewName
+      (updatePayload : any), // props
+    );
+  },
+
+  createInstance(
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+    hostContext : string | null,
+    internalInstanceHandle : Object
+  ) : Instance {
+    const tag = ReactNativeTagHandles.allocateTag();
+    const viewConfig = ReactNativeViewConfigRegistry.get(type);
+
+    if (__DEV__) {
+      for (let key in viewConfig.validAttributes) {
+        if (props.hasOwnProperty(key)) {
+          deepFreezeAndThrowOnMutationInDev(props[key]);
+        }
+      }
+    }
+
+    const updatePayload = ReactNativeAttributePayload.create(
+      props,
+      viewConfig.validAttributes
+    );
+
+    UIManager.createView(
+      tag, // reactTag
+      viewConfig.uiViewClassName, // viewName
+      rootContainerInstance, // rootTag
+      (updatePayload : any), // props
+    );
+
+    const component = new NativeHostComponent(tag);
+
+    precacheFiberNode(internalInstanceHandle, tag);
+
+    return component;
+  },
+
+  createTextInstance(
+    text : string,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : TextInstance {
+    const tag = ReactNativeTagHandles.allocateTag();
+
+    UIManager.createView(
+      tag, // reactTag
+      'RCTRawText', // viewName
+      rootContainerInstance, // rootTag
+      {text: text} // props
+    );
+
+    precacheFiberNode(internalInstanceHandle, tag);
+
+    return tag;
+  },
+
+  finalizeInitialChildren(
+    parentInstance : Instance,
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+  ) : void {
+    // Map from child objects to native tags.
+    // Either way we need to pass a copy of the Array to prevent it from being frozen.
+    const nativeTags = parentInstance._children.map(
+      (child) => typeof child === 'number'
+        ? child // Leaf node (eg text)
+        : child._nativeTag
+    );
+
+    UIManager.setChildren(
+      parentInstance._nativeTag, // containerTag
+      nativeTags // reactTags
+    );
+  },
+
+  getChildHostContext(
+    parentHostContext : string | null,
+    type : string
+  ) {
+    return parentHostContext;
+  },
+
+  insertBefore(
+    parentInstance : Instance | Container,
+    child : Instance | TextInstance,
+    beforeChild : Instance | TextInstance
+  ) : void {
+    const children = (parentInstance : any)._children;
```

Comment:
This is unsound if it ever is the container. Can we at least throw intentionally so it's clearly unimplemented?